### PR TITLE
Add docs for nginx config generator

### PR DESCRIPTION
**Pull request by jolynch**
https://github.com/jolynch/synapse-nginx

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -138,13 +172,19 @@ Each value in the services hash is also a hash, and must contain the following k
...
 
-The services hash *should* contain a section on how to configure the routing
-component you wish to use for this particular service. The only choice currently
-is `haproxy`:
+The services hash *should* contain a section on how to configure each routing
+component you wish to use for this particular service. The current choices are
+`haproxy` but you can access others e.g. [`nginx`](https://github.com/jolynch/synapse-nginx)
+through [plugins](createconfig). Note that if you give a routing component at the top level
+but not at the service level the default is typically to make that service
+available via that routing component, sans listening ports, if you wish to only
```
Comment (jnb) [inline README.md]: ... san listening ports.  If you wish to only...

File: README.md [hunk 2]
```diff
@@ -138,13 +172,19 @@ Each value in the services hash is also a hash, and must contain the following k
...
+configure a single component explicitly pass the ``disabled`` option to the
```
Comment (jnb) [inline README.md]: This didn't make sense to me.  Is it worth giving a more concrete example?

File: README.md [hunk 3]
```diff
@@ -28,15 +28,30 @@ Synapse solves these difficulties in a simple and fault-tolerant way.
 
 ## How Synapse Works ##
 
-Synapse runs on your application servers; here at Airbnb, we just run it on every box we deploy.
-The heart of synapse is actually [HAProxy](http://haproxy.1wt.eu/), a stable and proven routing component.
+Synapse typically runs on your application servers, often every machine. At the heart of Synapse
+are proven routing components like [HAProxy](http://haproxy.1wt.eu/) or [NGINX](http://nginx.org/),
+so you know that you won't drop a packet.
```
Comment (jolynch) [inline README.md]: sure

File: README.md [hunk 4]
```diff
@@ -28,15 +28,30 @@ Synapse solves these difficulties in a simple and fault-tolerant way.
...
+`config_generators` for configuring local state (e.g. load balancer configs)
+based on that service discovery state.
+
+Synapse supports service discovery with with pluggable `service_watcher`s.
+The service watchers take care of signaling to the `config_generators` so that
+they can react and reconfigure to point at available servers.
+
 We've included a number of default watchers, including ones that query zookeeper and ones using the AWS API.
-It is easy to write your own watchers for your use case, and we encourage submitting them back to the project.
+It is easy to write your own watchers for your use case, and install them as gems that
```
Comment (jolynch) [inline README.md]: I haven't written one yet ... but like etcd would be an example.

File: README.md (hunk 2 above)
Comment (jolynch) [inline README.md]: I think it's sorta confusing, can iterate with more examples later.

File: README.md [hunk 5]
```diff
@@ -416,19 +464,30 @@ frontend shared-frontend
...
+<a name="createconfig"/>
 ### Creating a Config Generator ###
 
 See the Config Generator [README](lib/synapse/config_generator/README.md) for
 how to add new Config Generators
+
+<a name="plugins"/>
+## Links to Synapse Plugins ##
+* [`synapse-nginx`](https://github.com/jolynch/synapse-nginx) Is a `config_generator`
+  which allows Synapse to automatically configure and administrate a local NGINX
```
Comment (jolynch) [inline README.md]: changed to administer.

**Pull request merged**